unlock-app/src/__tests__/components/interface/FinishSignup.test.tsx

```diff
@@ -43,4 +45,54 @@ describe('FinishSignup', () => {
       })
     )
   })
+
+  it('but not if the password does not match its confirmation', () => {
+    expect.assertions(1)
+
+    const emailAddress = 'geoff@bitconnect.gov'
+    const password = 'guest'
+
+    const { getByLabelText, getByDisplayValue } = rtl.render(
+      <FinishSignup
+        emailAddress={emailAddress}
+        signupCredentials={signupCredentials}
+      />
+    )
+
+    const inputs = [
+      getByLabelText('Password'),
+      getByLabelText('Confirm Password'),
+    ]
+    const submit = getByDisplayValue('Submit')
+
+    inputs.forEach((input, i) => {
+      rtl.fireEvent.change(input, { target: { value: password + i } })
+    })
+
+    rtl.fireEvent.click(submit)
+    expect(signupCredentials).not.toHaveBeenCalled()
+  })
+
+  describe('validatePassword', () => {
+    it('should not allow blank passwords', () => {
+      expect.assertions(1)
+      expect(validatePassword('', '').isValid).toBeFalsy()
+    })
+
+    it('should not allow passwords that are different from their confirmations', () => {
+      expect.assertions(1)
+      expect(validatePassword('guest', 'ghost').isValid).toBeFalsy()
+    })
+  })
+
+  describe('mapDispatchToProps', () => {
+    it('maps the dispatch to the props', () => {
+      expect.assertions(1)
+      expect(mapDispatchToProps((arg: any) => arg)).toEqual(
+        expect.objectContaining({
+          signupCredentials: expect.any(Function),
```

Comment:
can also add an assertion that calls the signupCredentials and verifies it spits out the right action too